sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -521,7 +521,15 @@ class BulkWriter(container: CosmosAsyncContainer,
 
           log.logInfo(s"invoking bulkInputEmitter.onComplete(), Context: ${operationContext.toString} ${getThreadInfo}")
           semaphore.release(activeTasks.get())
-          bulkInputEmitter.tryEmitComplete()
+          val completeEmitResult = bulkInputEmitter.tryEmitComplete()
+          if (completeEmitResult eq Sinks.EmitResult.OK) {
+            log.logDebug(s"bulkInputEmitter sink completed, Context: ${operationContext.toString} ${getThreadInfo}")
+          }
+          else {
+            log.logInfo(
```

Comment:
How about warning level?

Reply:
What's the impact of !OK status? How will it impact regular flow?

Reply:
Info is intentional - there are race conditions which can lead this to be completed already (in which case it would not be OK - and warning would be too noisy). INFO level logging usually is enabled on Spark environments -s o this seems reasonable.